apps/server/src/modules/tool/controller/mapper/tool-reference.mapper.ts

```diff
@@ -0,0 +1,19 @@
+import { ContextExternalToolDO, ExternalToolDO, ToolConfigurationStatus, ToolReference } from '@shared/domain';
+
+export class ToolReferenceMapper {
+	static mapToToolReference(
+		externalTool: ExternalToolDO,
+		contextExternalTool: ContextExternalToolDO,
+		status: ToolConfigurationStatus
+	): ToolReference {
+		const toolReference = new ToolReference({
+			contextToolId: contextExternalTool.id as string,
```

Comment:
The as string should not nessasary and if it is possible null, or undefined it should not wrong cast. 
If you really need this cast then you should add a method that check if id is a string. "tryCastToString()" with throw if not a string.